### ISSUE TITLE
Install and configure CELL OAI-PMH module

### ISSUE THREAD

**Issue by elmcip** [elmcip/elmcip#112]
We need to install and test the module that will enable the CELL search engine. It also requires configuration for field mapping. 
- Export path: /data/work/oai-pmh
- This is for the Creative Work (Machine name: work) content type only.
## Scope of work
- [x] Enable https://github.com/whiskyechobravo/cell_oai_pmh
- [x] Enable https://www.drupal.org/project/views_base_url
- [x] Convert dates stored inside field_story to a real date field (field_work_year). Commit b0e54e672b9405250cb030174313ebeec709b121 takes care of this. Also changes the needed views.
- [x] Create the needed mappings to get the ball rolling.
## Exported fields
- Author  field_author_name Entity Reference
- Contributor field_contributor Entity Reference
- Translator  field_translator  Entity Reference
- Year  field_number  Integer Text field  
- Publisher field_node_ref  Entity Reference
- Publication Type  field_venue_tax Term reference  Select list 
- Platform/Software field_work_platform_software  Entity Reference
- URL field_url Link
- Download URL  field_downloadurl Link
- Archive URL field_archiveurlLink  
- WorldCat Permalink  field_worldcat_link Link
- ISBN  field_isbn_text Text  Text field  
- Language  field_language_tax  Term reference  Select list 
- License field_license List (text) Select list 
- Event field_node_ref_event  Entity Reference
- Tags  taxonomy_vocabulary_1 Term reference  Autocomplete term widget (tagging)  
- Description (in English)  field_textfield Long text Text area (multiple rows) 
- Description (in original language)  field_abstract_lang Long text Text area (multiple rows) 
- Language of abstract  field_orig_lang List (text) Select list 
- Language of description (Taxonomy)  field_abstract_lang_tax Term reference  Select list 
- Contributors note field_contributors  Long text Text area (multiple rows) 
- Other language version  field_other_lang_ver_link Entity Reference
- Other ion field_other_ion_work  Entity Reference
- Part of another work  field_part_of_another_work  Entity Reference
- Pull Quotes field_pullquote Long text Text area (multiple rows) 
- Technical notes field_tech_notes  Long text Text area (multiple rows) 
- Appears in  field_in_anthology  Entity Reference
- Screen shots  field_multi_image Image Image 
- Multimedia  field_media_asset Multimedia asset (deprecated) Media file selector 
- Attachment  field_files File  File  
- Electronic Literature Directory entry field_electronic_literature_direLink  
- I ♥ E-Poetry Entry  field_i_e_poetry_entryLink  
- NT2 entry field_nt2_entry Link
- Po.Ex entry field_po_ex_entry Link
- Repository  field_repositoryLink  
- ELMCIP Anthology Submission field_anthology_submission  Boolean Single on/off checkbox  
- ELMCIP Anthology Selection  field_anthology_selection Boolean Single on/off checkbox  
- Record Status field_record_status_tax Term reference  Select list 
- KB or Notes field_orial_notes Text  Text field

A separate issue will need to be set up for bringing in new taxonomies for the CELL metadata.
### Installation

The Drupal OAI-PMH module is available for download to the CELL partners using Drupal 7 (any version). As a reminder, this module allows Drupal users to generate a marked up data flow using Dublin Core (which is mandatory for the [OAI-PMH](https://en.wikipedia.org/wiki/Open_Archives_Initiative_Protocol_for_Metadata_Harvesting) norms) and MODS (the standard that was agreed upon for this project). This module will allow Drupal users to map their data (the selected fields shared with all partners) with the correct semantic standards.

Listed below are the important elements to consider for the module configuration and implementation:
- A link to download the module: https://github.com/whiskyechobravo/views_oai_pmh
- Installation and configuration guidelines: you can download it via the same link.
- The field mapping document. It was made by the NT2 and revised by Brown University. The module needs to be configured according to this document.
  https://docs.google.com/spreadsheets/d/1bno53vunVujA-nxIWz9a4WebH9qvBjO5StD43xC2gdM/edit?usp=sharing

Please note that we had to rewrite the module completely in order to meet the CELL's specific requirements. We are now waiting on the approval of the original developers to submit the module on the drupal.org website. Ultimately, our version of the module would replace the original one. If that is the case, the CELL would be listed as the module sponsor, which is an interesting way to promote the project to the Drupal community.
## How to test

During development will the development branch be staged at http://test.elmcip.net/data/work/oai-pmh Point your scraper to this address and test it out.
